### PR TITLE
Remove fixed seed for test_sparse_nd_save_load

### PR DESCRIPTION
**Pull request by kalyc**
## Description ##
Remove fixed seed for test_sparse_nd_save_load
Unable to reproduce flakiness of test - ran 10000 times

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] Remove fixed seed for test_sparse_ndarray:test_sparse_nd_save_load

## Comments ##
- Related issue - https://github.com/apache/incubator-mxnet/issues/11741

@haojin2 @szha

### CODE REVIEW

**Comment by haojin2** [comment]
For this kind of case please refer my changes to #11888 and add the link to the tracking issue in the code so that we can backtrack if it happens to fail again.

File: tests/python/unittest/test_sparse_ndarray.py [hunk 1]
```diff
@@ -534,7 +534,9 @@ def test_sparse_nd_pickle():
                 assert same(a.asnumpy(), b.asnumpy())
 
 
-@with_seed(0)
+# @kalyc: Getting rid of fixed seed as flakiness could not be reproduced
+# tracked at https://github.com/apache/incubator-mxnet/issues/11741
+@with_seed()
```
Comment (szha) [inline tests/python/unittest/test_sparse_ndarray.py]: no need to add comment if not flaky.

Comment (kalyc) [reply tests/python/unittest/test_sparse_ndarray.py]: See comments by @haojin2 above

Comment (szha) [reply tests/python/unittest/test_sparse_ndarray.py]: I see

**Pull request merged**